dll/win32/shell32/shlexec.cpp

```diff
@@ -802,6 +802,15 @@ static UINT SHELL_FindExecutable(LPCWSTR lpPath, LPCWSTR lpFile, LPCWSTR lpVerb,
         xlpFile[0] = '\0';
     }
 
+  /* Clear any trailing periods */
+  for (INT i = wcslen(xlpFile); i >= 0; i--)
+  {
```

Comment:
```suggestion
  for (SIZE_T i = wcslen(xlpFile); i > 0; )
  {
      --i;
```

Reply:
Changed ```INT``` to ```SIZE_T``` and FOR test from ```i >= 0``` to ```i > 0```. Thanks.
In my testing if we move the FOR ```i--``` to a new line as you suggest the test failures go back to 36 from 35.
This is the case if we use ```--i;``` or ```i--;```. Was there a good reason to make this into another line?
For this reason, I have left the completion test of the FOR as ```i--``` as I had previously. Thanks.

Reply:
I wonder whether both the leading/trailing whitespace and dots could be more elegantly removed by using [`PathCanonicalizeW`](https://learn.microsoft.com/en-us/windows/win32/api/shlwapi/nf-shlwapi-pathcanonicalizew)

Reply:
@HBelusca, I have reviewed the examples within the link that you posted above and I do not see any way to make this work for me here. You are welcome to make specific code suggestions, but otherwise I cannot see how to do this. Thanks for the suggestion though.